src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -338,27 +338,39 @@ void SQLiteDBEngine::selectData(const std::string& table,
     }
 }
 
-void SQLiteDBEngine::deleteTableRowsData(const std::string& table,
-                                         const nlohmann::json& data)
+void SQLiteDBEngine::deleteTableRowsData(const std::string&    table,
+                                         const nlohmann::json& jsDeletionData)
 {
     if (0 != loadTableData(table))
     {
-        std::vector<std::string> primaryKeyList;
-        if (getPrimaryKeysFromTable(table, primaryKeyList))
+        const auto& itData{ jsDeletionData.find("data")};
+        const auto& itFilter{ jsDeletionData.find("row_filter_opt")};
+
+        if(itData != jsDeletionData.end() && itData->size() > 0)
```

Comment:
is neccesary the itData->size() ?

Reply:
I just add this check for taking into account the following case scenario:
```
{
  "table":"processes",
  "query":
  {
    "data":[],
    "row_filter_opt":"WHERE tid>=103"
  }
}
```
Here "data" exist but the size is 0.